src/main/scala/parsing/Parser.scala

```diff
@@ -36,20 +37,20 @@ object Parser extends Encoding {
           parsedLines = droppedLines.map(_.merge),
           droppedLines = parsedLines.map(_.merge)
         )
-      case ReaderType(input)   =>
-        val source: Source =  Source.fromString(input.toString)
-        val sourceLines: Iterator[String] = source.getLines()
-        val parsed: Seq[Either[EncodeType, EncodeType]] = sourceLines.toList.map(parsingLogic)
-        source.close()
+      case ReaderType(input) =>
+        val iterator = Iterator.continually(input.read().toChar).toArray
+        val lines  = Source.fromChars(iterator)
+        lines.close()
+        val parsed = lines.getLines().map(parsingLogic)
```

Comment:
close the source only after the getLines()

Reply:
done , thank you